client/src/test/java/org/eclipse/hono/client/impl/HonoClientImplTest.java

```diff
@@ -311,6 +311,61 @@ public void testOnRemoteCloseTriggersReconnection(final TestContext ctx) {
         // THEN the client invokes the disconnect handler provided in the original connect method call
         disconnectHandlerInvocation.await(500);
     }
+    
+    /**
+     * Verifies that it fails to connect after client was shutdown.
+     * 
+     * @param ctx The test context.
+     *
+     */
+    @Test
+    public void testConnectFailsAfterShutdown(final TestContext ctx) {
+
+        // GIVEN a shutdown client
+        ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
+        HonoClientImpl client = new HonoClientImpl(vertx, connectionFactory);
+        client.shutdown();
+        
+        // WHEN client gets connects 
+        Async connectionHandlerInvocation = ctx.async();
+        Handler<AsyncResult<HonoClient>> connectionHandler = result -> {
+            if(result.failed()) {
+                connectionHandlerInvocation.complete();
+            }
+        };
+        client.connect(new ProtonClientOptions(), connectionHandler );
+        
+        //THEN connect fails
+        connectionHandlerInvocation.await(500);
+    }
+    
+    /**
+     * Verifies that the client does not try to re-connect to a server instance if the client was shutdown.
+     * 
+     * @param ctx The test context.
+     *
+     */
+    @Test
+    public void testDoesNotTriggerReconnectionAfterShutdown(final TestContext ctx) {
```

Comment:
This test does not test the expected behavior FMPOV (and it also fails on my machine, supporting my assumption)